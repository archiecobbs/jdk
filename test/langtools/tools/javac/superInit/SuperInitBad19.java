/*
 * @test /nodynamiccopyright/
 * @bug 8194743
 * @summary Test order in which duplicate assignments are reported
 *
 * @compile/fail/ref=SuperInitBad19.out -XDrawDiagnostics SuperInitBad19.java
 */
class SuperInitBad19 {

    static class Example1 {

        final int x;

        {
            this.x = 123;       // this is the duplicate assignment
        }

        Example1(int x) {
            this.x = 456;
            super();
        }
    }

    static class Example2 {

        final int x;

        {
            this.x = 123;
        }

        Example2(int x) {
            super();
            this.x = 456;       // this is the duplicate assignment
        }
    }
}
