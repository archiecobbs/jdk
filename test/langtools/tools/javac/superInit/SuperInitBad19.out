SuperInitBad19.java:15:17: compiler.err.var.might.already.be.assigned: x
SuperInitBad19.java:34:17: compiler.err.var.might.already.be.assigned: x
2 errors
